explorer/ui/src/containers/DagContainer.ts

```diff
@@ -181,16 +181,34 @@ export class DagContainer {
                   }
                   remainingBlocks.splice(0, 0, block!);
                   runInAction(() => {
-                    this.blocks = remainingBlocks;
+                    this.blocks = observable.array(remainingBlocks, { deep: true });
                   });
                 }
               }
             } else if (event.hasNewFinalizedBlock()) {
-              this.errors.capture(
-                this.casperService.getLastFinalizedBlockInfo().then(block => {
+              const directFinalizedBlockHash = event.getNewFinalizedBlock()!.getBlockHash_asB64();
+
+              let finalizedBlocks = new Set(event.getNewFinalizedBlock()!.getIndirectlyFinalizedBlockHashesList_asB64());
+              finalizedBlocks.add(directFinalizedBlockHash);
+
+              let updatedLastFinalizedBlock = false;
+              this.blocks?.forEach(block => {
+                let bh = block.getSummary()!.getBlockHash_asB64();
+                if (finalizedBlocks.has(bh)) {
+                  block.getStatus()?.setIsFinalized(true);
+                }
+                if (!updatedLastFinalizedBlock && bh === directFinalizedBlockHash) {
```

Comment:
```suggestion
                if (bh === directFinalizedBlockHash) {
```
Doesn't look like that `!updatedLastFinalizedBlock` did anything, at most 1 block will match the `directFinalizedBlockHash` and that's when the flag is set to `true`.

Reply:
Maybe, when there are 10+ blocks needed to finalize a block, so that no block will match the `directFinalizedBlockHash`.